scalding-spark/src/main/scala/com/twitter/scalding/spark_backend/Op.scala

```diff
@@ -138,15 +138,14 @@ object Op extends Serializable {
     r.asInstanceOf[RDD[A]]
 
   final case class Transformed[Z, A](input: Op[Z], fn: RDD[Z] => RDD[A]) extends Op[A] {
-    @transient private val cache = new FutureCache[SparkSession, RDD[_ <: A]]
+    @transient private val cache = new FutureCache[SparkSession, RDD[_ <: A], Promise, Future]
```

Comment:
if we make my change to FutureCacheGeneric and leave FutureCache this diff should go away i think.